pkg/testutils/testutils.go

```diff
@@ -208,3 +211,40 @@ func EnsureServiceHealthy(t testing.TB, grpcAddr, httpAddr string, transportCred
 		require.Equal(t, http.StatusOK, resp.StatusCode, "unexpected status code received from server")
 	}
 }
+
+// MustDefaultConfig returns default server config with the playground, tracing and metrics turned off.
+func MustDefaultConfig() *config.Config {
+	config := config.DefaultConfig()
+
+	config.Playground.Enabled = false
+	config.Metrics.Enabled = false
+
+	return config
+}
+
+// MustDefaultConfigWithRandomPorts returns default server config but with random ports for the grpc and http addresses
+// and with the playground, tracing and metrics turned off.
+// This function may panic if somehow a random port cannot be chosen.
+func MustDefaultConfigWithRandomPorts() *config.Config {
+	config := MustDefaultConfig()
+
+	httpPort := TCPRandomPort()
+	grpcPort := TCPRandomPort()
+
+	config.GRPC.Addr = fmt.Sprintf("0.0.0.0:%d", grpcPort)
+	config.HTTP.Addr = fmt.Sprintf("0.0.0.0:%d", httpPort)
+
+	return config
+}
+
+// TCPRandomPort tries to find a random TCP Port. If it can't find one, it panics.
+// Else, it returns the port and releases the port.
+func TCPRandomPort() int {
+	l, err := net.Listen("tcp", "")
+	if err != nil {
+		panic(err)
+	}
+	addr := l.Addr().(*net.TCPAddr).Port
+	defer l.Close()
```

Comment:
Question: Will we have a case where we use the same port?  For example, we have 
```
port1 := TCPRandomPort()
port2 := TCPRandomPort()
```

Internally, it is possible that when we call TCPRandomPort the first time, the listener finds a free port (says 4567).  By the end of this function, 4567 is released back to the TCP listener since l.Close().  The next time TCPRandomPort is called, there is nothing to stop 4567 from being offer as a "free" port even it is not really free.

Reply:
ahhh yes 🤦‍♀️  good point, will revert this change